src/lib.rs

```diff
@@ -433,4 +515,179 @@ mod tests {
 
         assert_eq!(expected_tokens.to_string(), actual_tokens.to_string());
     }
+
+    #[test]
+    fn put_variants_in_module() {
+        let ast: DeriveInput = parse_quote! {
+            #[evt(module = "example")]
+            pub enum MyEnum {
+                A,
+                B
+            }
+        };
+
+        let actual_tokens = enum_variant_type_impl(ast);
+        let expected_tokens = quote! {
+            pub mod example {
+                use super::MyEnum;
+
+                pub struct A;
+
+                impl core::convert::From<A> for MyEnum {
+                    fn from(variant_struct: A) -> Self {
+                        MyEnum::A
+                    }
+                }
+
+                impl core::convert::TryFrom<MyEnum> for A {
+                    type Error = MyEnum;
+                    fn try_from(enum_variant: MyEnum) -> Result<Self, Self::Error> {
+                        if let MyEnum::A = enum_variant {
+                            core::result::Result::Ok(A)
+                        } else {
+                            core::result::Result::Err(enum_variant)
+                        }
+                    }
+                }
+
+                pub struct B;
+
+                impl core::convert::From<B> for MyEnum {
+                    fn from(variant_struct: B) -> Self {
+                        MyEnum::B
+                    }
+                }
+
+                impl core::convert::TryFrom<MyEnum> for B {
+                    type Error = MyEnum;
+                    fn try_from(enum_variant: MyEnum) -> Result<Self, Self::Error> {
+                        if let MyEnum::B = enum_variant {
+                            core::result::Result::Ok(B)
+                        } else {
+                            core::result::Result::Err(enum_variant)
+                        }
+                    }
+                }
+            }
+        };
+
+        assert_eq!(expected_tokens.to_string(), actual_tokens.to_string());
+    }
+
+    #[test]
+    fn derive_traits_for_all_variants() {
+        let ast: DeriveInput = parse_quote! {
+            #[evt(derive(Debug))]
+            pub enum MyEnum {
+                A,
+                B
```

Comment:
probably worth adding an inner derive to `B`, and asserting we have both `#[derive(Debug)]` and `#[derive(Clone)]` or something. we don't have to be clever and merge them into `#[derive(Clone, Debug)]` -- i don't think people will mind having

```rust
#[derive(Debug)]
#[derive(Clone)]
pub struct B;
```

generated, and there's no technical disadvantage

Reply:
Yep that's a great thing to add into the tests! And that's actually the currently implementation to *stack* the derives. I learnt when adding this feature that rust allows for this and handles having attributes with the same path on the same item which makes this alot easier :)